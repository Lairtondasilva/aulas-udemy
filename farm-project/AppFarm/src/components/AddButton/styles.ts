import { styled, css } from "styled-components/native";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";

export const Container = styled.View`
  ${({ theme }) => css`
    width: ${wp(20)}px;
    height: ${wp(20)}px;
    border-radius: ${wp(10)}px;
    background-color: #8b9f69;
    justify-content: center;
    align-items: center;
    position: absolute;
    bottom: 12px;
    right: 12px;
  `}
`;
