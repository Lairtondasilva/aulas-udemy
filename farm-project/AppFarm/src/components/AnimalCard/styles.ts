import { styled, css } from "styled-components/native";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";
import { Dimensions } from "react-native";

const { height } = Dimensions.get("window");

export const Container = styled.View`
  ${({ theme }) => css`
    display: flex;
    width: ${wp(37)}px;
    height: ${height > 684 ? hp(22) : hp(24)}px;
    justify-content: center;
    align-items: center;
    border-radius: 10px;
    border: 1px solid #cebebe;
    background: #fff;
    border-radius: 10px;
    elevation: 20;
    gap: ${hp(0.3)}px;
  `}
`;

export const containerImage = styled.ImageBackground`
  width: ${wp(30)}px;
  height: ${height > 684 ? hp(12) : hp(14)}px;
`;

export const Text = styled.Text`
  font-size: 24px;
  font-weight: bold;
`;
