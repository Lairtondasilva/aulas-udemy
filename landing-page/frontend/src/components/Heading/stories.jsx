import Heading from '.';

export default {
  title: 'Heading',
  component: Heading,
  args: {
    children: 'This is a heading',
  },
  argTypes: {
    children: { type: 'string' },
  },
};

export const Light = (args) => <Heading {...args} />;
export const Dark = (args) => <Heading {...args} />;

Light.args = {
  children: 'O texto está escuro',
  isLight: true,
};

Light.parameters = {
  backgrounds: {
    default: 'light',
  },
};

Dark.args = {
  children: 'O texto está claro',
  isLight: false,
};

Dark.parameters = {
  backgrounds: {
    default: 'dark',
  },
};
