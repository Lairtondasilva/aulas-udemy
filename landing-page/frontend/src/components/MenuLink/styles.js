import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.small};
    display: block;
    text-decoration: none;
    padding: ${theme.spacings.medium};
    color: ${theme.colors.primary};
    position: relative;

    &::after {
      //& se refere a proprio componente parecido com this em uma classe.
      content: '';
      position: absolute;
      bottom: 0.76rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondary};
      transition: all 300ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
