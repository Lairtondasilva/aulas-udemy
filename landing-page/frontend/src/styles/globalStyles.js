import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`

  * {
    margin: 0;
    padding: 0;
  }

  body {
    font-size: 1.6rem;
    font-family: ${({ theme }) => theme.fonts.body};
  }

  html {
    font-size: 65.5%;
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: ${({ theme }) => theme.fonts.heading};
    font-weight: bold;
  }

`;
