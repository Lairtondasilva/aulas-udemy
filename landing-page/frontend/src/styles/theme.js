export const theme = {
  colors: {
    primary: '#0a1128',
    secondary: '#dc143c',
    white: '#FFFFFF',
  },
  fonts: {
    body: "'Open Sans', sans-serif",
    heading: "'Montserrat', serif",
    sizes: {
      xSmall: '0.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xLarge: '4.0rem',
      xxLarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
  },
  spacings: {
    xSmall: '0.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xLarge: '4.0rem',
    xxLarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
};
