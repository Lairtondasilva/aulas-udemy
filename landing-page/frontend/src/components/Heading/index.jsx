import * as Styled from './styles';
import P from 'prop-types';

const Heading = ({
  children,
  isLight = true,
  as = 'h1',
  size = 'xHuge',
  uppercase = false,
}) => {
  return (
    <Styled.Title
      islight={String(isLight)}
      as={as}
      size={size}
      uppercase={String(uppercase)}
    >
      {children}
    </Styled.Title>
  );
};

Heading.propTypes = {
  children: P.node.isRequired,
  isLight: P.bool,
  as: P.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),
  size: P.oneOf(['small', 'medium', 'large', 'huge', 'xHuge']),
  uppercase: P.bool,
};

export default Heading;
