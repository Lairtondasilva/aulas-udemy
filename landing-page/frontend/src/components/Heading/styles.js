import styled, { css } from 'styled-components';
import { theme } from '../../styles/theme';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  large: (theme) => css`
    font-size: ${theme.fonts.sizes.xLarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.huge};
    ${mediaFont(theme)}
  `,
  xHuge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.large};
    background-color: red;
  }
`;

const titleCase = (upperCase) => css`
  text-transform: ${upperCase === 'true' ? 'upperCase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, islight, size, uppercase }) => css`
    color: ${islight === 'true' ? theme.colors.primary : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)}
  `}
`;
