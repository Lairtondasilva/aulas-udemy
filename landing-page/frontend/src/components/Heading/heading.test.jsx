import { screen } from '@testing-library/react';
import Heading from '.';
import { renderTheme } from '../../styles/renderTheme';

describe('<Heading />', () => {
  it('Should render with default values', () => {
    const { debug } = renderTheme(<Heading> texto </Heading>);
    debug();
    const heading = screen.getByRole('heading', { name: 'texto' });
    expect(heading).toMatchSnapshot();
  });
});
