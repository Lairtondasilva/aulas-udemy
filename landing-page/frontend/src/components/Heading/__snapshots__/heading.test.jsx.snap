// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> Should render with default values 1`] = `
.c0 {
  color: #0a1128;
  font-size: 6.4rem;
  text-transform: none;
}

<h1
  class="c0"
  islight="true"
  uppercase="false"
>
   texto 
</h1>
`;
